### PR TITLE
Doc for setting up a new environment name

### PR DESCRIPTION
**Pull request by Yivan**
This PR is to give clues how to configure a new environment name.

Even if it follow symfony standard, i think it is worth mentionning this in the documentation to help people (and those comming from Pimcore 4 can get trapped in the test environment which has a special signification in Symfony).

I state too that some environment file need to be created in `pimcore/lib/Pimcore/Bundle/CoreBundle/Resources/config/pimcore/` which seems is not documented.

This PR was done because of those issues relating some difficulties to set a new environment:
https://github.com/pimcore/pimcore/issues/2265
https://github.com/pimcore/pimcore/issues/2346

### CODE REVIEW

File: doc/Development_Documentation/21_Deployment/03_Multi_Environment.md [hunk 1]
```diff
@@ -33,16 +33,62 @@ var/config/system.php
...
+You can need to configure a new environment name different than the existing ones, which respect Symfony convention: `dev`, `test` and `prod`.
+
+> **Note:** The default `test` environment should only be used for running test suite (like Travis).
+> It use the session.storage.mock_file which fakes sessions (as consequence, the administrator cannot log in Pimcore for instance)
+
+To configure a new environment, you need to manually create a YAML config file for the project in two places (you can start by copying config_dev.yml for instance in each of those):
+
+```
+app/config/pimcore/
+pimcore/lib/Pimcore/Bundle/CoreBundle/Resources/config/pimcore/
```
Comment (kkarkus) [inline doc/Development_Documentation/21_Deployment/03_Multi_Environment.md]: @Yivan does it mean I have to create config inside core dir?

Comment (mgomse) [inline doc/Development_Documentation/21_Deployment/03_Multi_Environment.md]: Changes in the core isn't good idea IMHO. Isn't it possible to do it in some other way?

Comment (dpfaffenbauer) [reply doc/Development_Documentation/21_Deployment/03_Multi_Environment.md]: Shouldn't be necessary at all, ```app/config/pimcore``` or ```src/AppBundle/Resources/config``` should be totally enough

Comment (maff) [reply doc/Development_Documentation/21_Deployment/03_Multi_Environment.md]: @Yivan the only necessary file is `app/config/config_<environment>.yml`. Please do not suggest adding/changing core files. IMHO the previous description was fine - I'd just add the note regarding the test environment.

**Comment by Yivan** [comment]
I totally agree, it is strange to have to change core files, but it was needed when i tried (see the monlog error here https://github.com/pimcore/pimcore/issues/2265 please).
I will try again with the last version to see if this problem always occurs.

Maybe there is some special configuration flags to add in the new Yaml file created for the new environment (`app/config/config_<environment>.yml`) so no more monolog error is triggered ?
Just copying `app/config/config_dev.yml` to `app/config/config_mynewenv.yml` don't work.

Maybe a file config_env.yml.sample could be added as a starting point ?

**Comment by Yivan** [comment]
I updated the documentation, so there is no more need to create yaml file in the core bundle.
Instead i added some exemple/hints, so developper can create new environment like dev or prod as starting base.

If all is ok, this can be merged, thanks.

File: doc/Development_Documentation/21_Deployment/03_Multi_Environment.md [hunk 2]
```diff
@@ -33,16 +33,126 @@ var/config/system.php
...
+				new DebugBundle(),
+				new WebProfilerBundle(),
+				new SensioDistributionBundle()
+			], 80);
+			// add generator bundle only if installed
+			if (class_exists('Sensio\Bundle\GeneratorBundle\SensioGeneratorBundle')) {
+				$collection->addBundle(
+					new SensioGeneratorBundle(),
+					80, // priority
+					['dev'] // will use the dev config files of the bundle
```
Comment (maff) [inline doc/Development_Documentation/21_Deployment/03_Multi_Environment.md]: This means the bundle will only be loaded in the `dev` environment and should be changed to `staging` in your example (comment should be removed or changed).

**Comment by maff** [comment]
@Yivan I restructured the core config files in #2400, so they are now easily includable from a custom env config. This avoids having to replicate all the config for a specific environment (see the comment on the PR). Could you update this PR accordingly?

**Comment by maff** [comment]
@Yivan any update on this?

**Comment by Yivan** [comment]
@maff I will soon update my PR before the end of the week, is it ok ? Thanks.

**Comment by Yivan** [comment]
@maff Code has just been updated, if all is ok you can merge it! Thanks.

File: doc/Development_Documentation/21_Deployment/03_Multi_Environment.md [hunk 3]
```diff
@@ -33,16 +33,64 @@ var/config/system.php
...
+				new DebugBundle(),
+				new WebProfilerBundle(),
+				new SensioDistributionBundle()
+			], 80);
+			// add generator bundle only if installed
+			if (class_exists('Sensio\Bundle\GeneratorBundle\SensioGeneratorBundle')) {
+				$collection->addBundle(
+					new SensioGeneratorBundle(),
+					80, // priority
+					['staging'] // will use the staging config files of the bundle
```
Comment (maff) [inline doc/Development_Documentation/21_Deployment/03_Multi_Environment.md]: @Yivan could you change this to something like `the bundle will only be loaded in the staging environment`? It does not actually change how config files are handled, just sets a filter when loading the bundle.

**Comment by Yivan** [comment]
@maff change on the comment has been done, all should be ok now.

**Comment by maff** [comment]
Thanks!

**Pull request merged**